### PR TITLE
Correct owner for static java members.

### PR DESCRIPTION
**Pull request by bishabosha**
Set privateWithin for package private definitions. Set flags for classes.

### CODE REVIEW

File: tasty-query/shared/src/main/scala/tastyquery/Symbols.scala [hunk 1]
```diff
@@ -135,6 +134,11 @@ object Symbols {
         assert(false, s"cannot access owner, ${this.name} is local or not declared within any scope")
     }
 
+    /** What is the closest enclosing package of this symbol? Returns this if this is a package */
+    private[tastyquery] final def closestPackage: PackageSymbol = this match
+      case pkg: PackageSymbol => pkg
+      case sym: Symbol        => sym.owner.nn.closestPackage // owner of regular symbols is never null
```
Comment (sjrd) [inline tasty-query/shared/src/main/scala/tastyquery/Symbols.scala:140]:
```suggestion
      case pkg: PackageSymbol => pkg
      case sym: TermOrTypeSymbol => sym.owner.closestPackage
```
`Symbol` is partitioned into `PackageSymbol` and `TermOrTypeSymbol`, and the latter's `owner` is known to be non-null. :)

File: tasty-query/shared/src/main/scala/tastyquery/Symbols.scala [hunk 2]
```diff
@@ -135,6 +134,11 @@ object Symbols {
         assert(false, s"cannot access owner, ${this.name} is local or not declared within any scope")
     }
 
+    /** What is the closest enclosing package of this symbol? Returns this if this is a package */
```
Comment (sjrd) [inline tasty-query/shared/src/main/scala/tastyquery/Symbols.scala:137]:
```suggestion
    /** The closest enclosing package of this symbol.
      * Returns this if this is a package.
      */
```
Don't frame Scaladoc comments as questions.

File: tasty-query/shared/src/test/scala/tastyquery/TypeSuite.scala [hunk 3]
```diff
@@ -376,10 +376,53 @@ class TypeSuite extends UnrestrictedUnpicklingSuite {
...
+
+  testWithContext("bag-of-java-definitions[static]") {
+    val BagOfJavaDefinitionsClassMod = ctx.findTopLevelModuleClass("javadefined.BagOfJavaDefinitions")
+    val javadefinedPackage = ctx.findPackage("javadefined")
+
+    def testDef(name: TermName)(op: munit.Location ?=> TermSymbol => Unit)(using munit.Location): Unit =
+      op(BagOfJavaDefinitionsClassMod.findNonOverloadedDecl(name))
+
+    testDef(name"STATIC_INT") { STATIC_INT =>
+      assert(STATIC_INT.declaredType.isRef(defn.IntClass))
```
Comment (sjrd) [inline tasty-query/shared/src/test/scala/tastyquery/TypeSuite.scala:402]: Add
```suggestion
      assert(!STATIC_INT.isAnyOf(Flags.Private | Flags.Protected))
      assert(STATIC_INT.privateWithin == None)
```
?

Same for `defaultInt` just below, as well as in the test `"bag-of-java-definitions"`.

**Review by sjrd — CHANGES_REQUESTED** [review]

File: tasty-query/shared/src/main/scala/tastyquery/Symbols.scala (hunk 1 above)
Comment (bishabosha) [reply tasty-query/shared/src/main/scala/tastyquery/Symbols.scala]: thanks

**Review by sjrd — APPROVED** [review]

**Pull request merged**